### PR TITLE
React Capstone Code Review 

### PR DESCRIPTION
**Pull request by MMhaidari**
# What is added to the project
- [ ] All functionality requested added to the project
- [ ] This app contains three pages Rocket page, Missions page, My Profile page 
- [ ] Data rendered on the screen is fetched from provided APIs
- [ ] All the functionality added from Kanban 
- [ ] All components have been tested and working fine
- [ ] Styling is added for the project
- [ ] Application is working

### CODE REVIEW

**Review by rustam-tolipov — COMMENTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that did not use the correct `Gitflow`. Please, learn more about it [here](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) and resubmit your code. Thank you@

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Nemwel-Boniface — COMMENTED** [review]
##  Hello  @MMhaidari  and @Tshobohwa  👋,

I am Nemwel your reviewer.


## _This is an additional review as a result of the second opinion request._

I have received your second opinion request and had a look at it.

I believe that the reviewer @Rustamxon7  might have been a little confused about the flow to be used for group capstone projects and for solo capstone projects.

1. For group capstone projects the correct flow for requesting reviews is Github flow which means dev merging to main. (In this case, your project used the correct flow.
2. For solo capstone projects the flow for requesting reviews is Gitflow which means feature branch merging to dev.

As technically the flow you used is correct, kindly feel free to request for a review in your dashboard and you will receive a review.

In case anything is not 100% clear tag me @Nemwel-Boniface so that I get the notification.

Keep rocking 🚀!

**Comment by MMhaidari** [comment]
# What is added to the project
- [ ]  All functionality requested added to the project
- [ ]  This app contains three pages Rocket page, Missions page, My Profile page
- [ ]  Data rendered on the screen is fetched from provided APIs
- [ ]  All the functionality added from Kanban
- [ ]  All components have been tested and working fine
- [ ]  Styling is added for the project
- [ ]  Application is working

note: please read the above comments

File: src/__test__/Rocket.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  test('renders correctly when rocket is not reserved', () => {
+    const newRocket = { ...rocket, reserved: false };
+    const tree = renderer
+      .create(<Rocket rocket={newRocket} handleReserveClick={() => {}} onCancelClick={() => {}} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (danmainah) [inline src/__test__/Rocket.test.js:28]: - [ ] @team great work on your tests, however, I didn't see tests for some components you have 9 components and I could only see tests for 4 components, also tests for pure functions in your redux files.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @MMhaidari and @Tshobohwa , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- The My Profile section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/__test__/Rocket.test.js (hunk 1 above)
Comment (MMhaidari) [reply src/__test__/Rocket.test.js:28]: @danmainah dear we have only four stand-alone in the component folder even if you count the files one by one there are only exist 5 js files in the component folder. would you please enlighten me and name those nine components individually ?

Comment (danmainah) [reply src/__test__/Rocket.test.js:28]: - [ ] @MMhaidari Yes in the component folder you have four, but also add the ones in the route folder, as you can see in your project description it says all components, I would suggest you ask for a second opinion if you feel you disagree with some of the changes requested

**Comment by medaminedev66** [comment]
Hello @MMhaidari  👋,

_This is an additional review as a result of the second opinion request._

Based on the project requirements, all components must be tested. @danmainah was correct about the requested change. It appears that only four components have been tested.
@danmainah meant that the components inside the "routes" folder, including MyProfilePage, NotMatch, and RocketPage, should be tested. Additionally, the App, MyProfile, and NavBar components should be tested as well.

Testing all the components in React is necessary to ensure that the application works as expected and to catch any potential issues or bugs before they are deployed to production. 

Keep rocking 🚀!

**Comment by MMhaidari** [comment]
# What is added
- [ ] Requested changes added as instructed 
- [ ] Tests added for all remaining components
- [ ] Styling changes added as desired

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @TEAM

Your project is complete! Great job implementing the changes requested 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request✔️ ✔️
- Great collaboration✔️ ✔️
- Nice Design✔️ ✔️
- Test ok ✔️ ✔️
- Professional presentation of work✔️ ✔️
- No linter errors✔️ ✔️

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**